### PR TITLE
Fixed ShareClient.GetStatistics() bug

### LINKED ISSUE #9186: [BUG] ShareClient.GetStatistics(); can't handle more bytes than can be stored in an Int.
**Describe the bug**
ShareClient.GetStatistics(); throws exception when ShareUsageBytes value coming back from the API is too large.

***Exception or Stack Trace***
System.OverflowException
  HResult=0x80131516
  Message=Value was either too large or too small for an Int32.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type) in /_/src/System.Private.CoreLib/shared/System/Number.Parsing.cs:line 1933
at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)   at System.Int32.Parse(String s, IFormatProvider provider)   at Azure.Storage.Files.Shares.Models.ShareStatistics.FromXml(XElement element)   at Azure.Storage.Files.Shares.FileRestClient.Share.GetStatisticsAsync_CreateResponse(Response response)   at Azure.Storage.Files.Shares.FileRestClient.Share.GetStatisticsAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, Nullable 1 timeout, Boolean async, String operationName, CancellationToken cancellationToken)   at Azure.Storage.Files.Shares.ShareClient.GetStatisticsInternal(Boolean async, CancellationToken cancellationToken)   at Azure.Storage.TaskExtensions.EnsureCompleted[T](Task 1 task)   at Azure.Storage.Files.Shares.ShareClient.GetStatistics(CancellationToken cancellationToken)   at UserQuery.Main() in C:\Users\xxxx\AppData\Local\Temp\LINQPad6\xxxxx\LINQPadQuery:line 6   at LINQPad.ExecutionModel.ClrQueryRunner.Run()   at LINQPad.ExecutionModel.Server.RunQuery(QueryRunner runner)

**To Reproduce**
- Utilize more space (bytes) in your storage account file share than can be stored in an Int variable type. 
- Call the `ShareClient.GetStatistics();` Sdk.
- Notice that the number returning from the Azure Storage API is larger than an Int can represent.
- See exception "Value was either too large or too small for an Int32"

***Code Snippet***
```
using System;
using Azure.Storage;
using Azure.Storage.Files.Shares;

namespace ConsoleApp5
{
    class Program
    {
        static void Main(string[] args)
        {
            var cs = @"your-connection-string-here";
            var shareName = @"your-file-share-name-here";

            ShareClient share = new ShareClient(cs, shareName);
            var stats = share.GetStatistics();
        }
    }
}

```

**Expected behavior**
You should get back a statistic data model that represents the number of bytes if you have more than the number that can be represented by an in. Consider using an UInt64 instead of Int in this class property: ShareStatistics.ShareUsageBytes

**Screenshots**
image
(=6068040432 Bytes - can't be represented as Int type)

**Setup (please complete the following information):**
 - OS: any
 - IDE : any (used both linqpad 6 and vs 2019 enterprise)
 - Version of the Library used: Azure.Storage.Files.Shares 12.0.0
 - If you need to capture the API call result, you can turn on Fiddler's reverse-proxy, and it will start showing the .net core API calls. See here: https://www.telerik.com/forums/traffic-not-capture-unless-net-core-sdk-installed

edit: updated suggested property type to UInt64 instead of Int64.

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/9186

### CODE REVIEW

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/src/Generated/FileRestClient.cs [hunk 1]
```diff
@@ -9080,7 +9080,7 @@ public partial class ShareStatistics
         /// <summary>
         /// The approximate size of the data stored in bytes, rounded up to the nearest gigabyte. Note that this value may not include all recently created or recently resized files.
         /// </summary>
-        public int ShareUsageBytes { get; internal set; }
+        public long ShareUsageInBytes { get; internal set; }
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Files.Shares/src/Generated/FileRestClient.cs]: One example of this situation from BCL is Array.Length vs Array.LongLength, I'm not sure that we should use the same naming pattern but just FYI.

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    /// Stats for the share.
+    /// </summary>
+    public partial class ShareStatistics
+    {
+
+
+        /// <summary>
+        /// Warning: Share usage may exceed int.MaxValue.  Use ShareUsageInBytes instead.
+        /// </summary>
+        public int ShareUsageBytes {
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs]: Nit - use `[EditorBrowsable(Never)]` here so folks don't even see this in intellisense.

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            get
+            {
+                if (ShareUsageInBytes > int.MaxValue)
+                {
+#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
+                    throw new OverflowException("ShareUsageInBytes exceeds int.MaxValue.  Use ShareUsageInBytes instead.");
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs]: Change to `"ShareUsageBytes exceeds int.MaxValue.  Use ShareUsageInBytes instead."` and move to Constants

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    /// <summary>
+    /// ShareModelFactory provides utilities for mocking.
+    /// </summary>
+    public static partial class ShareModelFactory
+    {
+        /// <summary>
+        /// Creates a new PermissionInfo instance for mocking.
+        /// </summary>
+        public static ShareStatistics ShareStatistics(int shareUsageBytes)
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs]: Nit - use [EditorBrowsable(Never)] here so folks don't even see this in intellisense.

File: sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTests.cs [hunk 5]
```diff
@@ -440,6 +441,30 @@ public async Task GetStatisticsAsync()
...
+        {
+            // Arrange
+            long size = 3 * (long)Constants.GB;
+            MockResponse mockResponse = new MockResponse(200);
+            mockResponse.SetContent($"﻿<?xml version=\"1.0\" encoding=\"utf-8\"?><ShareStats><ShareUsageBytes>{size}</ShareUsageBytes></ShareStats>");
+            ShareClientOptions shareClientOption = new ShareClientOptions()
+            {
+                Transport = new MockTransport(mockResponse)
+            };
+            ShareClient shareClient = new ShareClient(new Uri(TestConfigDefault.FileServiceEndpoint), shareClientOption);
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTests.cs]: This still needs to be instrumented for the sync/async to work correctly.

**Review by tg-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs]: fixed.

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs]: fixed.

File: sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs (hunk 4 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareStatistics.cs]: fixed.

File: sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTests.cs (hunk 5 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTests.cs]: fixed.

**Comment by kemmis** [comment]
Nice job team. I’m very grateful to have experienced this process. #PleaseHireMeMicrosoft 🤣

👏🙏

**Pull request merged**